bugsnag-android-core/src/main/java/com/bugsnag/android/ErrorStore.java

```diff
@@ -170,14 +174,23 @@ private void flushErrorReport(File errorFile) {
             Logger.warn("Could not send previously saved error(s)"
                 + " to Bugsnag, will try again later", exception);
         } catch (Exception exception) {
-            if (delegate != null) {
-                Error minimalError = generateErrorFromFilename(errorFile.getName());
+            deleteStoredFiles(Collections.singleton(errorFile));
```

Comment:
What error case is this `catch` block handling?

Reply:
Any unchecked exceptions which can be thrown from within the `try` block, e.g. in the unlikely case that an `IllegalStateException` is thrown from within `Report#toStream()`.